arm-notificationhubs/2016-03-01/swagger/notificationhubs.json

```diff
@@ -0,0 +1,1742 @@
+                }, {
+                    "name": "parameters",
+                    "in": "body",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/definitions/PolicykeyResource"
+                    },
+                    "description": "Parameters supplied to regenerate the NotificationHub Authorization Rule Key."
+                }, {
+                    "$ref": "#/parameters/ApiVersionParameter"
+                }, {
+                    "$ref": "#/parameters/SubscriptionIdParameter"
+                }],
+                "responses": {
+                    "200": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/ResourceListKeys"
+                        }
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NotificationHubs/namespaces/{namespaceName}/notificationHubs/{notificationHubName}/pnsCredentials": {
+            "post": {
+                "tags": [
+                    "NotificationHubs"
+                ],
+                "operationId": "NotificationHubs_GetPnsCredentials",
+                "description": "Lists the PNS Credentials associated with a notification hub .",
+                "parameters": [{
+                    "name": "resourceGroupName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The name of the resource group."
+                }, {
+                    "name": "namespaceName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The namespace name."
+                }, {
+                    "name": "notificationHubName",
+                    "in": "path",
+                    "required": true,
+                    "type": "string",
+                    "description": "The notification hub name."
+                }, {
+                    "$ref": "#/parameters/ApiVersionParameter"
+                }, {
+                    "$ref": "#/parameters/SubscriptionIdParameter"
+                }],
+                "responses": {
+                    "200": {
+                        "description": "",
+                        "schema": {
+                            "$ref": "#/definitions/NotificationHubResource"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "CheckAvailabilityParameters": {
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "Gets or sets name"
+                },
+                "location": {
```

Comment:
is location necessary at all? Does the service check the name availability across all the locations or a specified location?
Either the location should be required or it should be removed.